src/Sentry/Breadcrumb.cs

```diff
@@ -39,7 +50,17 @@ public sealed class Breadcrumb : IJsonSerializable
     /// Contains a sub-object whose contents depend on the breadcrumb type.
     /// Additional parameters that are unsupported by the type are rendered as a key/value table.
     /// </remarks>
-    public IReadOnlyDictionary<string, string>? Data { get; }
+    public IReadOnlyDictionary<string, string>? Data
+    {
+        get => _sendDefaultPii
+            ? _data
+            : _data?.ToDictionary(
+                x => x.Key,
+                x => x.Value?.RedactUrl()
+                )!
```

Comment:
Curious why do we have to specify that the redacted result is not null if the returned type is a nullable dictionary.

Reply:
That was because of the nullability on the extension method.  Now that's been updated, you can simplify the usage here to:

```csharp
x => x.Value.RedactUrl());
```